packages/modules/carlo_gavazzi/counter.py

```diff
@@ -41,7 +41,8 @@ def update(self):
         power_all = sum(powers)
         currents = [abs(val / 1000) for val in self.__tcp_client.read_input_registers(
             0x0C, [ModbusDataType.INT_32] * 3, unit=unit)]
-        frequency = self.__tcp_client.read_input_registers(0x33, ModbusDataType.FLOAT_16, unit=unit) / 10
+        # ModbusDataType.FLOAT_16 wird erst ab Python 3.6 unterstützt
+        frequency = self.__tcp_client.delegate.read_input_registers(0x33, 2, unit=unit)[0] / 10
```

Comment:
Das ist von der Bedeutung jetzt allerdings sehr anders.

Die alte Version liest einen Register und macht einen float draus.

Die neue Version liest zwei Register, verwirft den zweiten gelesenen Register und nimmt den ersten davon so wie er ist, was vermute ich dann als UINT16 gelesen wird (so definitiv sehe ich das in der Manual leider nicht). Aber ziemlich sicher kein float.

Sowas in der Richtung müsste meine ich dann tun:
```python
frequency = struct.unpack('>e', self.__tcp_client.delegate.read_input_registers(0x33, 1, unit=unit).encode())
```
Allerdings ohne Wissen und test, sondern nur von ein bißchen stöbern in der manual.

Vielleicht auch besser statt `FLOAT_16` aus dem enum zu entfernen einen Sonderfall machen.

Also statt aktuell:
```python
result = [getattr(decoder, t.decoding_method)() for t in types]
```
Dann so:
```python
result = [struct.unpack(">e", struct.pack(">H", decoder.decode_16bit_uint())) if t == ModbusDataType.FLOAT_16 else getattr(decoder, t.decoding_method)() for t in types]
```
(Völlig ungetestet, ist nur so "aus meiner Erwartung runtergeschrieben").
Dann muss man den Modulcode nicht damit behelligen.

Reply:
Ich habe in diesem PR den alten Code wieder genommen: In [readgavazzi.py] wurde das Register auch nicht decodiert. (https://github.com/snaptec/openWB/blob/51df319b5612f5c339d7486846e6977d849967d2/modules/bezug_carlogavazzilan/readgavazzi.py#L103)

Reply:
Ahso... Ja dann war das mit dem FLOAT16 wohl sowieso falsch... Oder hat das irgendwo bei einem Test funktioniert? Der alte Code hat dann wohl einfach ein INT16 gelesen. Das könntest du hier dann auch machen.

Reply:
Nein, ich kann es ja auch nicht testen.